akvo/rsr/spa/app/modules/editor/actions.js

```diff
@@ -24,38 +24,40 @@ export const setStatus = (publishingStatus) => (dispatch, getState) => {
     .catch((error) => { dispatch({ type: actionTypes.BACKEND_ERROR, error, response: error.response.data }) })
 }
 export const fetchFields = (sectionIndex, fields) => ({ type: actionTypes.FETCH_SECTION, sectionIndex, fields })
-export const fetchSetItems = (sectionIndex, setName, items) => ({ type: actionTypes.FETCH_SET_ITEMS, sectionIndex, setName, items})
+export const fetchSetItems = (sectionIndex, setName, items) => ({ type: actionTypes.FETCH_SET_ITEMS, sectionIndex, setName, items })
 export const addSetItem = (sectionIndex, setName, item) => (dispatch, getState) => {
-  dispatch({ type: actionTypes.ADD_SET_ITEM, sectionIndex, setName, item})
+  dispatch({ type: actionTypes.ADD_SET_ITEM, sectionIndex, setName, item })
   const setItems = get(getState().editorRdr[`section${sectionIndex}`].fields, setName)
   const itemIndex = setItems.length - 1
   api.post(getEndpoint(sectionIndex, setName), item, getTransform(sectionIndex, setName, 'request'), null, true)
-    .then(({ data: {id, periods}}) => { dispatch({ type: actionTypes.ADDED_SET_ITEM, sectionIndex, setName, id, itemIndex, periods }) })
+    .then(({ data: { id, periods } }) => { dispatch({ type: actionTypes.ADDED_SET_ITEM, sectionIndex, setName, id, itemIndex, periods }) })
     .catch((error) => {
       dispatch({ type: actionTypes.BACKEND_ERROR, error, sectionIndex, setName: `${setName}[${setItems.length - 1}]`, response: error.response ? error.response.data : error, statusCode: error.response.status })
       dispatch({ type: actionTypes.REMOVED_SET_ITEM, failedAdd: true, sectionIndex, setName, itemIndex: setItems.length - 1, skipValidation: true })
     })
 }
 export const editSetItem = (sectionIndex, setName, itemIndex, itemId, fields) => (dispatch) => {
+  const inputFields = fields?.disaggregationTargets?.length > 0
+    ? { ...fields, disaggregationTargets: fields.disaggregationTargets.filter(item => (item?.value)) }
```

Comment:
We don't actually want to be filtering in the case when `item.value` is `null`. We want to retain the null values to unset values which were previously set. If a target was incorrectly set before, sending the `null` value unsets it. Filtering out those values would mean that we are not able to unset those values any more. 


Also, it looks like requests from different places seem to be sending different data. (before this filtering code is run). This is what the data looks like when I try to set a target on Project 7908 > **Result 1** > Indicator 1 > Period 1.  See below for another different example of data on the same project. 

```json
{
	"disaggregation_targets": [
		{
			"dimension_value": 973,
			"id": 26742,
			"period": 383236,
			"value": null
		},
		{
			"dimension_value": 974,
			"id": 26743,
			"period": 383236,
			"value": null
		},
		{
			"dimension_value": 975,
			"id": 26744,
			"period": 383236,
			"value": null
		},
		{
			"dimension_value": 977,
			"id": 26745,
			"period": 383236
		},
		{
			"dimension_value": 976,
			"id": 26746,
			"period": 383236,
			"value": null
		},
		{
			"dimension_value": 978,
			"id": 26747,
			"period": 383236,
			"value": null
		},
		{
			"dimension_value": 979,
			"id": 26748,
			"period": 383236,
			"value": null
		},
		{
			"dimension_value": 980,
			"id": 26749,
			"period": 383236,
			"value": 61000
		},
		{
			"dimension_value": 981,
			"id": 26750,
			"period": 383236,
			"value": null
		},
		{
			"dimension_value": 982,
			"id": 26751,
			"period": 383236,
			"value": null
		},
		{
			"dimension_value": 983,
			"id": 26752,
			"period": 383236,
			"value": null
		},
		{
			"dimension_value": 984,
			"id": 26753,
			"period": 383236,
			"value": null
		}
	]
}
```

This is what the data looks like when I try to set a target on Project 7908 > **Result 2** > Indicator 1 > Period 1. 

```json
{
	"disaggregation_targets": [
		{
			"dimension_value": 977,
			"value": "2"
		},
		null,
		null,
		{
			"dimension_value": 980,
			"value": "3"
		}
	]
}
```